File: src/@types/recipe.ts
export interface Direction {
  id: string
  order: number
  image?: string
  description: string
}

export interface Ingridient {
  id: string,
  measure: {
    kind: string
    quantity: number
  },
  description: string
}

export interface Recipe {
  id: string
  name: string
  numberOfServing?: number
  cookDuration?: number
  category: string
  origin: string
  tags: string[]
  image?: string
  video: string
  ingredients: Ingridient[]
  directions: Direction[]
}

File: src-tauri/entity/src/recipe.rs
use sea_orm::entity::prelude::*;
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, PartialEq, Eq, DeriveEntityModel, Deserialize, Serialize)]
#[sea_orm(table_name = "recipes")]
pub struct Model {
    #[sea_orm(primary_key)]
    #[serde(skip_deserializing)]
    pub id: Uuid,
    pub name: String,
    pub number_of_serving: u16,
    pub cook_duration: u16,
    pub category: String,
    pub origin: String,
    pub video: Option<String>,
    pub image: Option<String>,
}

#[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
pub enum Relation {
    #[sea_orm(has_many = "super::ingredients::Entity")]
    Ingredients,
    #[sea_orm(has_many = "super::directions::Entity")]
    Directions,
}

impl Related<super::ingredients::Entity> for Entity {
    fn to() -> RelationDef {
        Relation::Ingredients.def();
        Relation::Directions.def()
    }
}

impl ActiveModelBehavior for ActiveModel {}
